cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -629,3 +629,8 @@ if(MSVC)
     set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${MSVC_LINKER_FLAGS}")
   endif()
 endif()
+
+#--------------------------------------------------------------------------------------
+if(ARROW_COMPUTE)
+  add_definitions(-DARROW_HAVE_COMPUTE_MODULE)
```

Comment:
The preferred way would to be let CMake generate it: https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/config.h.cmake#L37
(also, if you just pass the option here, it might not be taken up by third-party applications including Arrow)